### ISSUE TITLE
Queries can be out of sync with the parser

### ISSUE THREAD

**Issue by Eyenseo** [nvim-treesitter/nvim-treesitter#2119]
### Before reporting

- [X] I have updated my neovim version to latest _master_
- [X] I have updated my plugin to the latest version
- [X] I have run `:TSUpdate`
- [X] I have read the [troubleshooting section](https://github.com/nvim-treesitter/nvim-treesitter#troubleshooting)

### Describe the bug

I've noticed that the queries come from this repo while the parsers come from an external one. If now a parser maintainer updates the queries it'll be a braking change until this repo picks these changes up. I wonder what motivation is for this behaviour instead of just fetching the queries too from the remote repos? 

### To Reproduce

Update queries / highlights in a parser. Run the TSInstall/FromGrammer functions and notice that the changes to queries are ignored. 

### Expected behavior

If I (remove and) install an parser it's from one cohesive version (of the repo the parser is from)

### Output of `:checkhealth nvim-treesitter`

```markdown
doesn't apply



nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found  0.20.1 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v17.2.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (GCC) 11.1.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - ocaml_interface✓ ✓ ✓ . ✓ 
  - beancount      ✓ . ✓ . . 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - sparql         ✓ ✓ ✓ ✓ ✓ 
  - comment        ✓ . . . . 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - ocaml          ✓ ✓ ✓ . ✓ 
  - dot            ✓ . . . ✓ 
  - nix            ✓ ✓ ✓ . ✓ 
  - glsl           ✓ ✓ ✓ ✓ ✓ 
  - json           ✓ ✓ ✓ ✓ . 
  - pug            ✓ . . . ✓ 
  - ledger         ✓ . ✓ ✓ ✓ 
  - php            ✓ ✓ ✓ ✓ ✓ 
  - clojure        ✓ ✓ ✓ . ✓ 
  - supercollider  ✓ ✓ ✓ ✓ ✓ 
  - heex           ✓ . ✓ ✓ ✓ 
  - godot_resource ✓ ✓ ✓ . . 
  - typescript     ✓ ✓ ✓ ✓ ✓ 
  - fennel         ✓ ✓ . . ✓ 
  - query          ✓ ✓ ✓ ✓ ✓ 
  - cpp            ✓ ✓ ✓ ✓ ✓ 
  - vue            ✓ . ✓ ✓ ✓ 
  - latex          ✓ . ✓ . ✓ 
  - make           ✓ . . . ✓ 
  - rst            ✓ ✓ . . ✓ 
  - css            ✓ . ✓ ✓ ✓ 
  - glimmer        ✓ . . . . 
  - erlang         . . . . . 
  - regex          ✓ . . . . 
  - svelte         ✓ . ✓ ✓ ✓ 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - fortran        ✓ . ✓ ✓ . 
  - teal           ✓ ✓ ✓ ✓ ✓ 
  - java           ✓ ✓ . ✓ ✓ 
  - hjson          ✓ ✓ ✓ ✓ ✓ 
  - gomod          ✓ . . . . 
  - dart           ✓ ✓ . ✓ ✓ 
  - prisma         ✓ . . . . 
  - http           ✓ . . . ✓ 
  - bash           ✓ ✓ ✓ . ✓ 
  - llvm           ✓ . . . . 
  - pioasm         ✓ . . . ✓ 
  - yaml           ✓ ✓ ✓ ✓ ✓ 
  - json5          ✓ . . . ✓ 
  - vim            ✓ ✓ . . ✓ 
  - cmake          ✓ . ✓ . . 
  - zig            ✓ . ✓ ✓ ✓ 
  - julia          ✓ ✓ ✓ ✓ ✓ 
  - bibtex         ✓ . ✓ ✓ . 
  - r              ✓ ✓ . . . 
  - devicetree     ✓ ✓ ✓ ✓ ✓ 
  - turtle         ✓ ✓ ✓ ✓ ✓ 
  - gdscript       ✓ ✓ . . ✓ 
  - html           ✓ ✓ ✓ ✓ ✓ 
  - jsdoc          ✓ . . . . 
  - tsx            ✓ ✓ ✓ ✓ ✓ 
  - commonlisp     ✓ ✓ ✓ . . 
  - yang           ✓ . ✓ . . 
  - surface        ✓ . ✓ ✓ ✓ 
  - ql             ✓ ✓ . ✓ ✓ 
  - elixir         ✓ ✓ ✓ ✓ ✓ 
  - cuda           ✓ ✓ ✓ ✓ ✓ 
  - go             ✓ ✓ ✓ ✓ ✓ 
  - c_sharp        ✓ ✓ ✓ . ✓ 
  - kotlin         ✓ . . . ✓ 
  - ruby           ✓ ✓ ✓ ✓ ✓ 
  - fish           ✓ ✓ ✓ ✓ ✓ 
  - ocamllex       ✓ . . . ✓ 
  - scala          ✓ . ✓ . ✓ 
  - dockerfile     ✓ . . . ✓ 
  - rust           ✓ ✓ ✓ ✓ ✓ 
  - perl           ✓ . . . . 
  - tlaplus        ✓ . ✓ . ✓ 
  - jsonc          ✓ ✓ ✓ ✓ ✓ 
  - toml           ✓ ✓ ✓ ✓ ✓ 
  - graphql        ✓ . . ✓ ✓ 
  - scss           ✓ . . ✓ . 
  - hcl            ✓ . ✓ ✓ ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```
```


### Output of `nvim --version`

```text
doesn't apply


NVIM v0.6.0
Build type: Release
LuaJIT 2.0.5
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
```


### Additional context

_No response_

**Comment by theHamsta** [comment]
We had this discussions few times in the beginning of this project

- most of the parser queries are only sparsely maintained, often PRs are almost never merged or only in batches once a year
- The queries very not documented in the beginning of the project and even Atom did not use queries but their own mapping format for highlighting. This created a fragmentation of the queries present in different parser repos and the ones used in different editors. Now they are mostly inconsistent or sometimes even incompatible with each other. Though we tried to align with usage of the official parser
- The queries in most parser repos are mostly bare-bones. Since they weren't useful for rich highlighting we decided to do our own queries instead of taking the upstream ones.
- We can use Vim-specific captures and Lua captures when we have our own queries

Usually the parsers queries are out of date in the repos, while we got a huge community help to create consistent queries between all the languages. Helix for instance took our queries, emacs-tree-sitter took their own. It would be nice to share more queries in future with other projects or auto-import queries that maintain queries specifically for nvim-treesitter.

If your sorrow of out-dating is, that loading the queries may brake or we forget new nodes:

- every update to the lockfile.json is tested on CI (TSInstall only install tested and compatible updates from the upstream parsers)
- Look on the changes of the parsers, when we get an automated update-PR for the lockfile. When there are new nodes, we typically add missing queries. Most updates to the parsers, however, just use nodes that have been already existing before. It is of course always possible that we oversee something.
